library/system/src/lib/yast2/fs_snapshot.rb

```diff
@@ -0,0 +1,125 @@
+    CREATE_SNAPSHOT_CMD = "/usr/lib/snapper/installation-helper --step 5 --description \"%s\""
+    LIST_SNAPSHOTS_CMD = "LANG=en_US.UTF-8 /usr/bin/snapper --no-dbus list"
+
+    attr_reader :number, :snapshot_type, :previous, :timestamp, :user, :cleanup, :description
+
+    # Determines whether snapper is configured or not
+    #
+    # @return [true,false] true if it's configured; false otherwise.
+    def self.configured?
+      out = Yast::SCR.Execute(Yast::Path.new(".target.bash_output"), FIND_CONFIG_CMD)
+      out["exit"] == 0
+    end
+
+    # Configures snapper
+    #
+    # @return [true,false] true if it's configured; false otherwise.
+    def self.configure
+      if configured?
+        true
+      else
+        out = Yast::SCR.Execute(Yast::Path.new(".target.bash_output"), CREATE_CONFIG_CMD)
+        out["exit"] == 0
+      end
+    end
+
+    # Creates a new snapshot
+    #
+    # It raises and exception if Snapper is not configured.
+    #
+    # @param description [String] Snapshot's description.
+    # @return [FsSnapshot,nil] The created snapshot if the operation was
+    #                          successful. Otherwise, it returns nil.
+    def self.create(description)
+      raise SnapperNotConfigured unless configured?
+      out = Yast::SCR.Execute(Yast::Path.new(".target.bash_output"), CREATE_SNAPSHOT_CMD % description)
+      if out["exit"] == 0
+        find(out["stdout"].to_i) # The CREATE_SNAPSHOT_CMD returns the number of the new snapshot.
+      else
+        nil
+      end
+    end
+
+    # Returns all snapshots
+    #
+    # It raises and exception if Snapper is not configured.
+    #
+    # @return [Array<FsSnapshot>] All snapshots that exist in the system.
+    def self.all
+      raise SnapperNotConfigured unless configured?
+      out = Yast::SCR.Execute(Yast::Path.new(".target.bash_output"), LIST_SNAPSHOTS_CMD)
+      lines = out["stdout"].split(/\n/)[2..-1] # relevant lines from output.
+      lines.map do |line|
+        data = line.split(/\s*\|\s*/)
+        timestamp = data[3] == "" ? nil : DateTime.parse(data[3])
+        new(data[1].to_i, data[0].to_sym, data[2].to_i, timestamp, data[4],
+            data[5].to_s.to_sym, data[6])
+      end
+    end
+
+    # Finds an snapshot by its number
+    #
+    # It raises and exception if Snapper is not configured.
+    #
+    # @param nubmer [Fixnum] Number of the snapshot to search for.
+    # @return [FsSnapshot,nil] The snapshot with the number +number+ if found.
+    #                          Otherwise, it returns nil.
+    # @see FsSnapshot.all
+    def self.find(number)
+      all.find { |s| s.number == number }
+    end
+
+    def initialize(number, snapshot_type, previous, timestamp, user, cleanup, description)
+      @number = number
+      @snapshot_type = snapshot_type
+      @previous = previous
+      @timestamp = timestamp
+      @user = user
+      @cleanup = cleanup
+      @description = description
+    end
+  end
+end
```

Comment:
One more idea. We need more logging, ideally for all bash call to see what is called and what is result. It really helps with debugging on customer side when we do not know, what can go wrong. If we have logs we can clearly say - snapper return wrong value or it return correct data and problem is in our code.